modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/suspended/MonadSyntax.kt

```diff
@@ -0,0 +1,30 @@
+package arrow.typeclasses.suspended
+
+import arrow.Kind
+import arrow.typeclasses.Applicative
+import arrow.typeclasses.Monad
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+import kotlin.coroutines.startCoroutine
+
+interface MonadSyntax<F> : ApplicativeSyntax<F>, Monad<F> {
+  override fun <A> f(fa: suspend () -> A): Kind<F, A> =
```

Comment:
@JorgeCastilloPrz yes, each monad level understand how to inject a suspend function which is what denotes a side effect. 

@nomisRev `effect` is only available for `MonadDefer` and up. `effect` is the same as `::identity` for suspended functions and it's only available there because is the only place where you can truly suspend and bind side effects. `effect` returns `A` but `f` which is the same as `liftM` for which other versions called `liftM` are in this PR returns `Kind<F, A>`.

You need these two functions to go from side effects to wrapped pure values depending on how you assemble the composition of your side effects and if you go back and forth between `suspend () -> A` and `Kind<F, A>`.

We need to get together to better name all of these so they are friendlier to users.